harvester_e2e_tests/fixtures/api_client.py

```diff
@@ -198,6 +198,44 @@ def skip_version_after(request, api_client):
                 )
 
 
+@pytest.fixture(autouse=True)
+def skip_version_if(request, api_client):
+    ''' To mark test case should be skip when hit the condition string.
+
+    Args:
+        *args: Version string prefixing with one of operators: `!=`, `==`, `>=`, `<=`, `>`, `<`
+    Keyword Args:
+        reason: The reason string for `pytest.skip`, default is:
+            "Cluster Version `{cluster_version}` is not included in versions: {versions}"
+        condition: Condition callable function to check compare result(bool), default is `all`
+    '''
+    default_reason = (
+        "Cluster Version `{cluster_version}` is not included in versions: {versions}"
+    )
+    mark = request.node.get_closest_marker("skip_version_if")
+    if mark:
+        cluster_ver = api_client.cluster_version
+        checks = [version_check(vstr, cluster_ver) for vstr in mark.args]
+        reason = mark.kwargs.get('reason', default_reason)
+        if mark.kwargs.get('condition', all)(r for *_, r in checks):
+            versions = [f"{op} {v}" for op, v, _ in checks]
+            return pytest.skip(
+                reason.format(cluster_version=cluster_ver, versions=versions)
+            )
+
+
+def version_check(vstring, version):
+    from operator import le, lt, ge, gt, ne, eq
+    ops = {"<=": le, "<": lt, ">=": ge, ">": gt, "!=": ne, "==": eq}
+
+    try:
+        op = target_ver = None
+        op, target_ver = re.search(r"([<>=!]+)\s?(.+)", vstring).groups()
```

Comment:
Would you be able to leave a comment as to what's taking place in this `re.search`?
It's a tad difficult to parse :sweat_smile:

Reply:
we might not remove the marker before `v1.2` EOL (unless 4293 will have a backport fix).